src/main/java/org/jboss/windup/operator/controller/WindupDeploymentController.java

```diff
@@ -0,0 +1,68 @@
+package org.jboss.windup.operator.controller;
+
+import io.fabric8.kubernetes.api.model.OwnerReference;
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import io.fabric8.kubernetes.client.Watcher;
+import io.fabric8.kubernetes.client.dsl.MixedOperation;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import lombok.extern.java.Log;
+import org.apache.commons.lang3.StringUtils;
+import org.jboss.windup.operator.model.WindupResource;
+import org.jboss.windup.operator.model.WindupResourceDoneable;
+import org.jboss.windup.operator.model.WindupResourceList;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.inject.Named;
+
+@Log
+@ApplicationScoped
+public class WindupDeploymentController implements Watcher<Deployment> {
+    @Inject
+    MixedOperation<WindupResource, WindupResourceList, WindupResourceDoneable, Resource<WindupResource, WindupResourceDoneable>> crClient;
+
+    @Named("namespace")
+    String namespace;
+
+    @Override
+    public void eventReceived(Action action, Deployment deployment) {
+        log.info("Event " + action + " received for Deployment : " + deployment.getMetadata().getName());
+
+        updateCRStatus(deployment);
```

Comment:
_Comment to further review the code, NOT mandatory for this PR_
`eventReceived` implementation in `WindupController` is "Action aware" since we switch between different methods based on `Action action`.
Is it fine, in this case, to always run the `updateCRStatus` method, no matter the `action` value?

Reply:
Exactly , if it's an ADD, MODIFIED, DELETE it always will do an updateSRCStatus..... counting the status of the deployments
In the future it will have to take care of the result number of the pods in order to deploy/kill those that are  expected